### PR TITLE
(#268) - Fix 'App is broken when --sources and --target directories are same with --overwrite option is enabled'

### LINKED ISSUE #268: App is broken when --sources and --target directories are same with --overwrite option is enabled
App is broken when --sources and --target directories are same when --overwrite option is enabled. It deletes `--sources` directory also.

Example broken command

```
java -jar target/jpeek-jar-with-dependencies.jar --sources /Users/baybatu/workspace/zerocrat/jpeek/jpeek_tests --target /Users/baybatu/workspace/zerocrat/jpeek/jpeek_tests --overwrite
```
Gives following stack trace:

```
Directory /Users/baybatu/workspace/zerocrat/jpeek/jpeek_tests deleted
Exception in thread "main" java.lang.IllegalStateException: java.nio.file.NoSuchFileException: /Users/baybatu/workspace/zerocrat/jpeek/jpeek_tests
	at org.jpeek.skeleton.Classes.iterator(Classes.java:104)
	at org.cactoos.collection.CollectionOf.lambda$new$0(CollectionOf.java:77)
	at org.cactoos.scalar.UncheckedScalar.lambda$value$0(UncheckedScalar.java:58)
	at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:58)
	at org.cactoos.func.IoCheckedBiFunc.apply(IoCheckedBiFunc.java:66)
	at org.cactoos.func.UncheckedBiFunc.apply(UncheckedBiFunc.java:60)
	at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:59)
	at org.cactoos.scalar.UncheckedScalar.value(UncheckedScalar.java:59)
	at org.cactoos.collection.CollectionEnvelope.iterator(CollectionEnvelope.java:69)
	at org.cactoos.iterable.Mapped.lambda$null$0(Mapped.java:68)
	at org.cactoos.iterable.IterableEnvelope.iterator(IterableEnvelope.java:57)
	at org.cactoos.collection.CollectionOf.lambda$new$0(CollectionOf.java:77)
	at org.cactoos.scalar.UncheckedScalar.lambda$value$0(UncheckedScalar.java:58)
	at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:58)
	at org.cactoos.func.IoCheckedBiFunc.apply(IoCheckedBiFunc.java:66)
	at org.cactoos.func.UncheckedBiFunc.apply(UncheckedBiFunc.java:60)
	at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:59)
	at org.cactoos.scalar.UncheckedScalar.value(UncheckedScalar.java:59)
	at org.cactoos.collection.CollectionEnvelope.iterator(CollectionEnvelope.java:69)
	at org.cactoos.collection.CollectionEnvelope.iterator(CollectionEnvelope.java:69)
	at org.cactoos.scalar.AndInThreads.value(AndInThreads.java:303)
	at org.cactoos.scalar.AndInThreads.value(AndInThreads.java:66)
	at org.cactoos.scalar.UncheckedScalar.lambda$value$0(UncheckedScalar.java:58)
	at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:58)
	at org.cactoos.func.IoCheckedBiFunc.apply(IoCheckedBiFunc.java:66)
	at org.cactoos.func.UncheckedBiFunc.apply(UncheckedBiFunc.java:60)
	at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:59)
	at org.cactoos.scalar.UncheckedScalar.value(UncheckedScalar.java:59)
	at org.jpeek.skeleton.Skeleton.packages(Skeleton.java:144)
	at org.jpeek.skeleton.Skeleton.xml(Skeleton.java:107)
	at org.jpeek.App.analyze(App.java:149)
	at org.jpeek.Main.run(Main.java:151)
	at org.jpeek.Main.main(Main.java:115)
Caused by: java.nio.file.NoSuchFileException: /Users/baybatu/workspace/zerocrat/jpeek/jpeek_tests
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
	at java.nio.file.Files.readAttributes(Files.java:1737)
	at java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:219)
	at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
	at java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:322)
	at java.nio.file.FileTreeIterator.<init>(FileTreeIterator.java:72)
	at java.nio.file.Files.walk(Files.java:3574)
	at java.nio.file.Files.walk(Files.java:3625)
	at org.jpeek.DefaultBase.files(DefaultBase.java:63)
	at org.jpeek.skeleton.Classes.iterator(Classes.java:99)
	... 32 more

Process finished with exit code 1
```

It would be better to validate `--sources` and `--target` before delete `--target` directory and if they are equal, then show user-friendly/descriptive error message.

### PR DESCRIPTION
**Pull request by humb1t**
Fixes #268 - prevent equal source and target paths in case of `overwrite` option enabled.

### CODE REVIEW

**Comment by 0crat** [comment]
@humb1t[/z](https://www.0crat.com/u/humb1t) this pull request is too small, just 3 lines changed (less than 10), there will be no formal code review, see [§53](http://www.zerocracy.com/policy.html#53) and [§28](http://www.zerocracy.com/policy.html#28); in the future, try to make sure your pull requests are not too small; @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) please review this and merge or reject

**Comment by humb1t** [comment]
@paulodamaso - I think it's all.

I changed LoC coverage limit because even additional test was not enough to move forward and it out of scope to pull it up.

File: src/main/java/org/jpeek/Main.java [hunk 1]
```diff
@@ -125,12 +125,31 @@ public static void main(final String... args) throws IOException {
      * @throws IOException If fails
      */
     private void run() throws IOException {
-        final ConsoleAppender console = new ConsoleAppender();
+        if (this.overwrite && this.sources.equals(this.target)) {
+            throw new IllegalArgumentException(
+                "Invalid paths, source and target paths shouldn't be equal."
```
[the remark this answers is not in the document]
Comment (humb1t) [reply src/main/java/org/jpeek/Main.java]: @paulodamaso  thanks, I will change the error message. What about LoC limit in this issue and in #261, I not only tried on master, but also tried with new `git clone` and had the same result - here is an issue #320

**Comment by humb1t** [comment]
@0crat wait for #320 fix

**Comment by 0crat** [comment]
> @0crat wait for #320 fix ([here](https://github.com/yegor256/jpeek/issues/318#issuecomment-484977843))

@humb1t Job [#318](https://github.com/yegor256/jpeek/issues/318) is not in scope, can't put it on hold

File: src/main/java/org/jpeek/Main.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (humb1t) [reply src/main/java/org/jpeek/Main.java]: @paulodamaso - is that message better?

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@humb1t please take a look to the error message

**Comment by paulodamaso** [comment]
@humb1t rultor is triggering an error, please take a look

**Comment by 0crat** [comment]
Job `gh:yegor256/jpeek#318` is not assigned, can't get performer

**Comment by 0crat** [comment]
This job is not in scope

**Pull request closed without merging**